Iryna_Ramanets/Module_1_Java/Java_Fundamentals/HW4/src/com/epam/calculator/test/DivTest.java

```diff
@@ -0,0 +1,73 @@
+package com.epam.calculator.test;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+public class DivTest extends BaseTest {
+
+	@Test(description = "Simle positive long division:", dataProvider = "dataForDivLongPositive",
+			groups = {"positive"})
+	public void testDivLongPositive(long first, long second, long expected) {
+		long div = calculator.div(first, second);
+		Assert.assertEquals(div, expected, "");
+	}
+
+	@DataProvider(name = "dataForDivLongPositive")
+	public Object[][] dataForDivLongPositive() {
+		return new Object[][] {
+			{ 4, 2, 2 },
+			{ 12, 3, 4 },
+			{ 6, -3, -2 },
+			{ 0, 1, 0 },
+			{ -1, 1, 1 }
+		};
+	}
+
+	@Test(description = "Simle negative long division:", dataProvider = "dataForDivLongNegative",
+			expectedExceptions = NumberFormatException.class, groups = {"negative"})
+	public void testDivLongNegative(long first, long second) {
+		calculator.div(first, second);
+	}
+
+	@DataProvider(name = "dataForDivLongNegative")
+	public Object[][] dataForDivLongNegative() {
+		return new Object[][] {
+			{ 4, 0 }    // This is a BUG: no exception for div on 0 for long.
+		};
+	}
+	
+	@Test(description = "Simle positive double division:", dataProvider = "dataForDivDoublePositive",
+			groups = {"positive"})
+	public void testDivDoublePositive(double first, double second, double expected) {
+		double div = calculator.div(first, second);
+		Assert.assertEquals(div, expected, "");
```

Comment:
Assert.assertEquals() has few implementations with different count of input parameters. If you use the method that includes a message, please provide the Error message. Example: "Actual result(<value>) does not match expected(<value>)"
The same for all other similar cases.